packages/@lwc/template-compiler/src/codegen/codegen.ts

```diff
@@ -57,6 +62,13 @@ export default class CodeGen {
     inlineStyleImports: t.ImportDeclaration[] = [];
     inlineStyleBody: t.Statement[] = [];
 
+    generateKey() {
```

Comment:
This doesn't seem like the right place to generate keys. Maybe it should be moved to the runtime, although it would be nice to do this work before then during compile time.